### PR TITLE
moves setting of colors for views to middleware

### PR DESCRIPTION
**Pull request by emilyb7**
related #122

### CODE REVIEW

**Review by njsfield — APPROVED** [review]

**Review by njsfield — APPROVED** [review]

File: routes/middleware.js [hunk 1]
```diff
@@ -8,8 +8,13 @@ exports.initLocals = (req, res, next) => {
 
   const locals = res.locals;
   locals.user = req.user;
+  const colors = require('./setup/colors.js');
```
Comment (des-des) [inline routes/middleware.js]: names confusing.
```js
const viewColors = require('./setup/view_colors.js')
```

File: routes/middleware.js [hunk 2]
```diff
@@ -8,8 +8,13 @@ exports.initLocals = (req, res, next) => {
...
+  const route = req.url.split('/')[1];
```
Comment (des-des) [inline routes/middleware.js]:
```js
const viewName=
```

File: routes/middleware.js [hunk 3]
```diff
@@ -8,8 +8,13 @@ exports.initLocals = (req, res, next) => {
...
+  locals.color = Object.keys(colors).indexOf(route) > -1
```
Comment (des-des) [inline routes/middleware.js]: Okay this is a bit weird
```js
Object.keys(colors).indexOf(route) > -1 ?
```
is equivalent to
```js
colors[route] ?
```
anyway, with new names we come out with
```js
locals.color = viewColors[viewName] || viewColors.default
```
Which actually makes sense.. ?
or
```js
const viewColor = viewColors[viewName]
locals.color = viewColor !== indefined ? viewColor : viewColors.default
```
kinda more robust (empty string is falsy)

File: routes/setup/colors.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
```
Comment (des-des) [inline routes/setup/colors.js]: yh rename this file as above?

File: templates/views/director_profile.pug [hunk 5]
```diff
@@ -3,7 +3,7 @@ include ../mixins/snippets
 include ../mixins/movie-listing
 
 block content
-  #title-container.bg-green.pa3.shadow-5.z-1.light-gray
+  #title-container.pa3.shadow-5.z-1.light-gray(class=`bg-${color}`)
```
Comment (des-des) [inline templates/views/director_profile.pug]: Finally,

Can we change the name of this variable (locals.color) to something more specific?

Something like primaryColor ..?

**Review by des-des — CHANGES_REQUESTED** [review]

**Review by des-des — APPROVED** [review]

**Pull request merged**